### DISCUSSION TITLE
German version of the book

### DISCUSSION THREAD

**Discussion by S1lncer** [hendricius/the-sourdough-framework#87] [Q&A]
Hi there, will there be in the forseen future a german version of the book for readers that do not understand english?

**Comment by hendricius** [comment]
Maybe sometime in 2024. It sounds like it could be a lot of work to translate everything and keep the content in sync. @cedounet - you being the LaTeX expert, do you see any way how this could be done?

**Comment by cedounet** [comment]
On Wed, Aug 09, 2023 at 04:41:51AM -0700, Hendrik Kleinwaechter wrote:
> Maybe sometime in 2024. It sounds like it could be a lot of work to
> translate everything and keep the content in sync. @cedounet - you
> being the LaTeX expert, do you see any way how this could be done?

Not really beyond blood and sweat... Translating SW is fine as you only
translate short strings in a XML file through some UI.. The
documentation is more complicated: see 
https://wiki.gnome.org/TranslationProject/ContributeTranslations

1) You need to translate the bulk of it, that is just work, a lot of it
but not really difficult as long as you speak German :D

2) A you say once this is done, need to translate every single patch...
it gets tedious quickly.

3) LaTeX separate form and content but only so much... German texts tend
to be longer and we sometimes do manual adjustement either on the text
or by making suggestions to TeX on how to deal with a paragraph.  This
is where it gets tricky quickly.

4) you can write macro that helps and babel will provide the
infrastructure for words like "Inhaltsverzeichnis", dates, etc... 

Things like :
    \newcommand{\translation}[2]{#1}% use English version
    %\newcommand{\translation}[2]{#2}% use German version

    \translation{I like bread}{Ich mag Brot}

or better with \input{ch1.tex} but still too much work to my liking.
See above for gnome, you would need a UI that allows you to see both
text.  It surely exist, but no idea.

4) I am not even talking about pictures and graphs.

Best,

C/

**Comment by hendricius** [comment]
@cedounet thanks for the detailed evaluation of the possible translation. Based on the feedback I suggest we postpone this until way later, it sounds like a lot of work that will cause issues with every subsequent book change.
